promql/quantile.go

```diff
@@ -153,15 +153,23 @@ func bucketQuantile(q float64, buckets buckets) (float64, bool, bool) {
 
 // histogramQuantile calculates the quantile 'q' based on the given histogram.
 //
-// The quantile value is interpolated assuming a linear distribution within a
-// bucket.
-// TODO(beorn7): Find an interpolation method that is a better fit for
-// exponential buckets (and think about configurable interpolation).
+// For custom buckets, the result is interpolated linearly, i.e. it is assumed
+// the observations are uniformly distributed within each bucket. (This is a
+// quite blunt assumption, but it is consistent with the interpolation method
+// used for classic histograms so far.)
+//
+// For exponential buckets, the interpolation is done under the assumption that
+// the samples within each bucket are distributed in a way that they would
+// uniformly populate the buckets in a hypothetical histogram with higher
+// resolution. For example, if the rank calculation suggests that the requested
+// quantile is right in the middle of the population of the (1,2] bucket, we
+// assume the quantile would be right at the bucket boundary between the two
```

Comment:
Conversely if the schema is -1, then given this input:
```
clear
load 5m
    low_res_histogram {{schema:-1 sum:5 count:6 buckets:[0 6]}}
```

Which is:  (1,4] count =6. I would expect:
```
eval instant at 5m histogram_quantile(0.5, low_res_histogram)
    {}  2.0
```

But I get
```
error in eval histogram_quantile(0.5, low_res_histogram) (line 822): expected 2 for {} but got 2.2426406871192848
```

Reply:
I think you got a point here. The math should work out independent of the schema, but it doesn't. I have to re-think.

Reply:
OK, I realized the dead end I have manoeuvred myself into.